app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/CommonTriageQuestionsController.scala

```diff
@@ -419,12 +420,59 @@ class CommonTriageQuestionsController @Inject() (
               else routes.SingleDisposalsTriageController.whenWasCompletionDate()
           }
 
-        Ok(
-          previousReturnExistsWithSameCompletionDatePage(
-            state.fold(_.subscribedDetails, _.subscribedDetails).isATrust,
-            backLink
-          )
+        val invalidCompletionDate = state.fold(
+          _.newReturnTriageAnswers
+            .fold(
+              _.fold(
+                _.completionDate,
+                e => Some(e.completionDate)
+              ),
+              _.fold(
+                _.completionDate,
+                e => Some(e.completionDate)
+              )
+            ),
+          _.draftReturn
+            .triageAnswers()
+            .fold(
+              _.fold(
+                _.completionDate,
+                e => Some(e.completionDate)
+              ),
+              _.fold(
+                _.completionDate,
+                e => Some(e.completionDate)
+              )
+            )
```

Comment:
if `invalidCompletionDate` is `None` you can redirect to either the single or multiple disposal triage cya endpoint depending on what type of triage answers you have - if it's `None` then it means you shouldn't be on this page cos it means no completion date has been supplied to us.